File: app/src/main/java/wordmemory/idalavye/com/wordmemory/adapters/HomePagePagerAdapter.java
package wordmemory.idalavye.com.wordmemory.adapters;

import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;
import wordmemory.idalavye.com.wordmemory.fragments.homepage.ExercisesFragment;
import wordmemory.idalavye.com.wordmemory.fragments.homepage.WordsListingFragment;

public class HomePagePagerAdapter extends FragmentStatePagerAdapter {

    int mNumOfTabs;

    public HomePagePagerAdapter(FragmentManager fm, int NumOfTabs) {
        super(fm);
        this.mNumOfTabs = NumOfTabs;
    }

    @Override
    public Fragment getItem(int position) {
        switch (position) {
            case 0:
                WordsListingFragment wordsListingFragment = new WordsListingFragment();
                return wordsListingFragment;
            case 1:
                ExercisesFragment exercisesFragment = new ExercisesFragment();
                return exercisesFragment;
            default:
                return null;
        }
    }

    @Override
    public int getCount() {
        return mNumOfTabs;
    }
}

File: app/src/main/java/wordmemory/idalavye/com/wordmemory/adapters/HomePagePagerAdapter.kt
package wordmemory.idalavye.com.wordmemory.adapters

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import wordmemory.idalavye.com.wordmemory.fragments.homepage.ExercisesFragment
import wordmemory.idalavye.com.wordmemory.fragments.homepage.WordsListingFragment

class HomePagePagerAdapter(fragmentManager: FragmentManager, private val numOfTabs: Int) : FragmentStatePagerAdapter(fragmentManager) {
    override fun getItem(position: Int): Fragment? {
        return when (position) {
            0 -> WordsListingFragment()
            1 -> ExercisesFragment()
            else -> null
        }
    }

    override fun getCount(): Int {
        return numOfTabs
    }
}
